jax/experimental/shard_map.py

```diff
@@ -700,6 +701,15 @@ def _debug_callback_eager_rule(mesh, *args, callback: Callable[..., Any],
   return []
 eager_rules[debugging.debug_callback_p] = _debug_callback_eager_rule
 
+def _device_put_eager_rule(mesh, x, *, src, device):
```

Comment:
nit: src and device are reversed.

so it should be: `(mesh, x, *, device, src)`

Reply:
Thanks, but they're keyword-only, so the order doesn't matter, right?

Reply:
Maybe but I feel that maintaining the order is less confusing and it doesn't hurt to do that here :)